diff --git a/renderers/video_renderer.c b/renderers/video_renderer.c
--- a/renderers/video_renderer.c
+++ b/renderers/video_renderer.c
@@ -430,7 +430,10 @@ void video_renderer_init(logger_t *render_logger, const char *server_name, video
             gst_pipeline_use_clock(GST_PIPELINE_CAST(renderer_type[i]->pipeline), clock);
             renderer_type[i]->appsrc = gst_bin_get_by_name (GST_BIN (renderer_type[i]->pipeline), "video_source");
             g_assert(renderer_type[i]->appsrc);
-            g_object_set(renderer_type[i]->appsrc, "caps", caps, "stream-type", 0, "is-live", TRUE, "format", GST_FORMAT_TIME, NULL);
+            g_object_set(renderer_type[i]->appsrc, "caps", caps, "stream-type", 0,
+                         "is-live", TRUE, "format", GST_FORMAT_TIME,
+                         "do-timestamp", rtp ? TRUE : FALSE,
+                         NULL);
             g_string_free(launch, TRUE);
             gst_caps_unref(caps);
             gst_object_unref(clock);
